ui/de-lib/src/main/java/org/iplantc/de/analysis/client/presenter/sharing/AnalysisSharingPresenter.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * 
+ * @author sriram
+ * 
+ */
+
+package org.iplantc.de.analysis.client.presenter.sharing;
+
+import org.iplantc.de.analysis.client.views.sharing.AnalysisSharingView;
+import org.iplantc.de.client.models.analysis.Analysis;
+import org.iplantc.de.client.models.diskResources.PermissionValue;
+import org.iplantc.de.client.models.sharing.SharedResource;
+import org.iplantc.de.client.models.sharing.Sharing;
+import org.iplantc.de.client.services.AnalysisServiceFacade;
+import org.iplantc.de.client.sharing.SharingPermissionsPanel;
+import org.iplantc.de.client.sharing.SharingPresenter;
+import org.iplantc.de.client.util.JsonUtil;
+import org.iplantc.de.collaborators.client.util.CollaboratorsUtil;
+
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.user.client.ui.HasOneWidget;
+
+import com.sencha.gxt.core.shared.FastMap;
+
+import java.util.List;
+
+public class AnalysisSharingPresenter implements SharingPresenter {
+
+    final AnalysisSharingView view;
+    private final SharingPermissionsPanel permissionsPanel;
+    private final List<Analysis> selectedAnalysis;
+    private Appearance appearance;
+    private FastMap<List<Sharing>> dataSharingMap;
+    private FastMap<List<JSONObject>> sharingList;
+    private final JsonUtil jsonUtil;
+    private final CollaboratorsUtil collaboratorsUtil;
+
+    public AnalysisSharingPresenter(final AnalysisServiceFacade aService,
```

Comment:
I presume it has to do with the `TODO`s but a few of the variables defined here are not used.